packages/flutter/test/material/time_picker_test.dart

```diff
@@ -124,6 +124,87 @@ void main() {
     );
   });
 
+  testWidgets('Material2 - show 24 Hour Format', (WidgetTester tester) async {
+    addTearDown(tester.view.reset);
+
+    // When use24HourFormat: false, should show AM/PM indicators
+    await pumpShowTimePicker24HourFormat(tester, materialType: MaterialType.material2, use24HourFormat: false);
+    expect(find.text(amString), findsOneWidget);
+    expect(find.text(pmString), findsOneWidget);
+    await tester.tap(find.text(okString)); // Dismiss the dialog
+    await tester.pumpAndSettle();
+
+    // When use24HourFormat: true, should not show AM/PM indicators
+    await pumpShowTimePicker24HourFormat(
+      tester,
+      materialType: MaterialType.material2,
+    );
+    expect(find.text(amString), findsNothing);
+    expect(find.text(pmString), findsNothing);
+
+    // When use24HourFormat: true, hours should be displayed in 24 hour format
+    final List<String> labels00To22 = List<String>.generate(12, (int index) {
```

Comment:
Should this call labels00To23?

Reply:
here, we are multiplying indexes from 0 to 11 with 2, so it would be 00to22, right?